src/base/string.h

```diff
@@ -94,29 +94,41 @@ inline std::string as_string(const char* str) {
 inline std::string as_string(const int8_t t) {
```

Comment:
Why do we need to change this? is int16_t etc. defined by boost?

Reply:
No, all those are built-in types defined by the C++ standard. Unfortunately some compilers have trouble casting some "normal" types like `unsigned long` to a fixed-width type. So if we call `as_string(some_vector.size())`, then some compilers can not decide whether to call the `uint16_t`, `uint32_t`, `uint64_t`, `int32_t`, or `int64_t` overload and abort compilation. So we instead need to provide overloads for all of `short`, `int`, `long`, and `long long`, each in their `signed` and `unsigned` variant  – all compilers seem to be happy to resolve this then.